core/templates/pages/profile-page/profile-page.component.spec.ts

```diff
@@ -0,0 +1,288 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for profile page component.
+ */
+
+// TODO(#7222): Remove the following block of unnecessary imports once
+// thread-data.service.ts is upgraded to Angular 8.
+import { UpgradedServices } from 'services/UpgradedServices';
+
+require('pages/profile-page/profile-page.component.ts');
+
+describe('Profile page', function() {
+  var $scope = null;
+  var ctrl = null;
+  var $httpBackend = null;
+  var $q = null;
+  var UserService = null;
+  var CsrfTokenService = null;
+  var DateTimeFormatService = null;
+  var $log = null;
+  var windowRefMock = {
+    nativeWindow: {
+      location: {
+        href: '',
+        reload: function() {}
+      }
+    }
+  };
+
+  beforeEach(angular.mock.module('oppia', $provide => {
+    let ugs = new UpgradedServices();
+    for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
+      $provide.value(key, value);
+    }
+  }));
+  beforeEach(angular.mock.module('oppia', function($provide) {
+    $provide.value('UrlService', {
+      getUsernameFromProfileUrl: () => 'username1'
+    });
+    $provide.value('WindowRef', windowRefMock);
+  }));
+  beforeEach(angular.mock.inject(function($injector, $componentController) {
+    $httpBackend = $injector.get('$httpBackend');
+    $q = $injector.get('$q');
+    UserService = $injector.get('UserService');
+    CsrfTokenService = $injector.get('CsrfTokenService');
+    DateTimeFormatService = $injector.get('DateTimeFormatService');
+    $log = $injector.get('$log');
+
+    spyOn(CsrfTokenService, 'getTokenAsync')
+      .and.returnValue($q.resolve('sample-csrf-token'));
+
+    var $rootScope = $injector.get('$rootScope');
+    $scope = $rootScope.$new();
+    ctrl = $componentController('profilePage', {
+      $scope: $scope
+    });
+  }));
+
+  afterEach(function() {
+    windowRefMock.nativeWindow.location.href = '';
+  });
+
+  it('should get locate date string', function() {
+    // This corresponds to Fri, 21 Nov 2014 09:45:00 GMT.
+    var NOW_MILLIS = 1416563100000;
+    spyOn(DateTimeFormatService, 'getLocaleDateString').withArgs(NOW_MILLIS)
```

Comment:
I don't think you should be mocking this function... That defeats the purpose of the test

Reply:
How should I test this function so? Without the mock, the tests would fail because the date-time format changes depending of the computer. That's why I've mocked it to be sure it won't fail.

Reply:
Hmmm, I see.